src/commands/setup.ts

```diff
@@ -0,0 +1,55 @@
+import { Firestore, WriteBatch } from '@google-cloud/firestore';
+import { Client, Message } from 'discord.js';
+
+exports.run = async (client: Client, msg: Message, args: string[], firestore: Firestore) => {
+    if (msg.member!.permissions.has('ADMINISTRATOR') ) { // Ensures only admins may use this command
+        const guildRef = firestore.collection('guilds').doc(msg.guild!.id);
+
+        const batchArray: WriteBatch[] = [];
+        batchArray.push(firestore.batch());
+        let operationCount = 0;
+        let batchIndex = 0;
+
+        // Setup Firestore batches into array
+        msg.guild!.members.forEach((member) => {
+            if (!member.user.bot) {
```

Comment:
Same here about early exists, though I'm not sure what happens when you return inside a js foreach